src/streams.h

```diff
@@ -385,7 +385,7 @@ class CDataStream
     {
         // Special case: stream << stream concatenates like stream += stream
         if (!vch.empty())
-            s.write((char*)&vch[0], vch.size() * sizeof(vch[0]));
+            s.write((char*) vch.data(), vch.size() * sizeof(typename decltype(vch)::value_type));
```

Comment:
In commit "Fix subscript-0 in streams.h"

It would seem more direct (and shorter) to write sizeof(*vch.data()).